lib/Exception/DocumentManagerException.php

```diff
@@ -11,6 +11,59 @@
 
 namespace Sulu\Component\DocumentManager\Exception;
 
+/**
+ * Domain exception for the document manager.
+ */
 class DocumentManagerException extends \Exception
 {
+    private $documentManagerName;
+
+    /**
+     * @param string $message
+     * @param string $documentManagerName
+     * @param \Exception $previousException
+     */
+    public function __construct($message, $documentManagerName = null, \Exception $previousException = null)
+    {
+        parent::__construct($message, null, $previousException);
+
+        if ($documentManagerName) {
+            $this->formatMessage($documentManagerName);
+        }
+    }
+
+    /**
+     * Set the name of the document manager which threw this exception.
+     *
+     * @param string $name
+     */
+    public function setDocumentManagerName($name)
```

Comment:
Because the thing that threw this exception probably does not know which document manager invoked it. This is why we catch the Exceptions in the DocumentManager - to set the name.